vtr_flow/scripts/python_libs/vtr/util.py

```diff
@@ -0,0 +1,515 @@
+
+        cmd_errored = (cmd_returncode != expected_return_code)
+
+        #Send to stdout
+        if self._show_failures and (self._verbose or (cmd_errored and self._verbose_error)):
+            for line in cmd_output:
+                print (indent_depth*self._indent + line,)
+
+
+        if (self._show_failures and cmd_errored):
+            raise CommandError(msg="Executable {exec_name} failed".format(exec_name=PurePath(orig_cmd[0]).name), 
+                               cmd=cmd,
+                               log=str(PurePath(temp_dir).joinpath(log_filename)),
+                               returncode=cmd_returncode)
+
+        return cmd_output, cmd_returncode
+
+    def check_command(self, command):
+        """
+        Return True if command can be run, False otherwise.
+        """
+
+        #TODO: actually check for this
+        return True
+
+
+def write_tab_delimitted_csv(filepath, rows):
+    """
+    Write out the data provied in a tab-delimited CSV format
+
+    filepath: The filepath to write the data to
+    rows: An iterable of dictionary-like elements; each element 
+          provides a set key-value pairs corresponding to a row 
+          in the output file
+    """
+    #Calculate the max width of each column
+    columns = OrderedDict()
+    for row in rows:
+        for key, value in row.items():
+
+            if key not in columns:
+                columns[key] = max(len(key), len(str(value)))
+            else:
+                columns[key] = max(columns[key], len(str(value)))
+
+    #Write the elements
+    with open(filepath, 'w+') as f:
+        writer = csv.writer(f, delimiter='\t')
+
+        #Write out the header
+        header = []
+        for col_name, col_width in columns.items():
+            header.append("{:{width}}".format(col_name, width=col_width))
+        writer.writerow(header)
+
+        #Write rows
+        for row in rows:
+            values = []
+            for col_name, col_width in columns.items():
+                values.append("{:{width}}".format(row[col_name], width=col_width))
+            writer.writerow(values)
+
+def load_tab_delimited_csv(filepath):
+
+    data = []
+    with open(filepath) as f:
+        reader = csv.reader(f, delimiter='\t')
+
+        header = None
+        for csv_row in reader:
+            if header == None:
+                header = [val.strip() for val in csv_row]
+            else:
+                data_row = OrderedDict()
+
+                for i, value in enumerate(csv_row):
+                    data_row[header[i]] = value.strip()
+
+                data.append(data_row)
+
+    return data
+
+def make_enum(*sequential, **named):
+    """
+    Makes an enumeration
+
+    >> MY_ENUM = enum(ONE, TWO, THREE)
+    """
+    enums = dict(zip(sequential, range(len(sequential))), **named)
+    reverse = dict((value, key) for key, value in enums.items())
+    enums['reverse_mapping'] = reverse
+    return type('Enum', (), enums)
+
+def mkdir_p(path):
```

Comment:
pathlib.mkdir(parents=True) should work here.  We don't need this function.

Reply:
mkdir_p has been removed and replaced with pathlib.mkdir